### PR TITLE
OCPBUGS-29479: fix asynccache improper initialization

### PR DESCRIPTION
**Pull request by stlaz**
There were two bugs in the async cache initialization:
1. the initialization never returns an error even if it fails to run the caching function
2. The context in the caching function is cancelled right after the polling returns. If the cached item makes use of the context (OIDC providers unfortunately do), it will start failing.

/assign @TheRealJon 
/cc @jhadvig

### CODE REVIEW

File: pkg/serverutils/asynccache/asynccache.go [hunk 1]
```diff
@@ -33,20 +32,31 @@ func NewAsyncCache[T any](ctx context.Context, reloadPeriod time.Duration, cachi
...
-				return false, nil
+		true,
+		func(pollingCtx context.Context) (bool, error) {
+			itemChan := make(chan struct{}, 1)
+			var item T
+			var err error
+			go func() {
+				item, err = cachingFunc(ctx)
+				itemChan <- struct{}{}
+			}()
```
Comment (TheRealJon) [inline pkg/serverutils/asynccache/asynccache.go:47]: Wondering, could we use channels to pass both item and error through to be handled below:
```suggestion
		func(pollingCtx context.Context) (bool, error) {
			itemChan := make(chan T, 1)
			errChan := make(chan error, 1)
			go func() {
				i, e := cachingFunc(ctx)
				if e != nil {
				  errChan <- e
				  return
				}
				itemChan <- i
			}()
```

**Review by TheRealJon — COMMENTED** [review]
Just one comment, let me know what you think.

File: pkg/serverutils/asynccache/asynccache.go (hunk 1 above)
Comment (stlaz) [reply pkg/serverutils/asynccache/asynccache.go:47]: We could although I am not sure it's necessary at this moment. The channel simply serves as a notifier that a result is available.

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Pull request merged**